File: upl-purchase-sdk-php/src/Model/Currency.php
<?php

/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */

declare(strict_types=1);

namespace Unzer\PayLater\Model;

use InvalidArgumentException;

use function array_key_exists;
use function sprintf;

class Currency implements Enum
{
    public const EUR = 'EUR';
    public const CHF = 'CHF';
    public const USD = 'USD';
    public const CAD = 'CAD';
    public const GBP = 'GBP';

    public const ALLOWED_TYPES = [
        'EUR' => self::EUR,
        'CHF' => self::CHF,
        'USD' => self::USD,
        'CAD' => self::CAD,
        'GBP' => self::GBP,
    ];

    /**
     * @var string
     */
    private $value;

    /**
      * @param string $value
      * @throws InvalidArgumentException
      */
    public function __construct(string $value)
    {
        if (!array_key_exists($value, self::ALLOWED_TYPES)) {
            throw new InvalidArgumentException(sprintf('Unexpected value "%s"', $value));
        }
        $this->value = $value;
    }

    /**
     * @return string
     */
    public function getValue(): string
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->value;
    }
}

File: upl-purchase-sdk-java/src/main/java/com/unzer/paylater/model/Currency.java
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */
package com.unzer.paylater.model;

public enum Currency {

    EUR("EUR"),
    CHF("CHF"),
    USD("USD"),
    CAD("CAD"),
    GBP("GBP");

    private String value;

    Currency(String value) {
        this.value = value;
    }

    public static Currency fromValue(String value) {
        for (Currency b : Currency.values()) {
            if (b.value.equals(value)) {
                return b;
            }
        }
        throw new IllegalArgumentException("Unexpected value '" + value + "'");
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return String.valueOf(value);
    }
}

